starter-code/src/contact.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from "react";
+import './App.css';
+import contacts from './contacts.json';
+
+class Contact extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            contact: contacts.sort(() => Math.random() - 0.5),
+            clickCount: 5
+        }
+    }
+    clickHandler = () => {
+        this.setState({
+            clickCount: this.state.clickCount + 1
+        })
+    }
+    sortByName = () => {
+        this.setState({
+            contact: contacts.sort((a, b) => a.name > b.name)//Ne pas mettre de slice à cet endroit, sinon on bloque l'array à une certaine taille
+        })
+    }
+
+    sortByPopularity = () => {
+        this.setState({
+            contact: contacts.sort((a, b) => a.popularity > b.popularity)
+        })
+    }
+
```

Comment:
The logic is there but I have the impression that your SortByPopularity() is not working. Try doing something like this when sorting numbers:
```js
function compareNumbers(a, b) {
  return a - b;
}
```